src/scenic/domains/driving/roads.py

```diff
@@ -352,6 +352,8 @@ class Road(LinearElement):
     laneGroups: Tuple[LaneGroup] = None
     sections: Tuple[RoadSection]    # sections in order from start to end
 
+    signals: Tuple[Signal]
```

Comment:
Where are these signals? Are they all signals encountered at either end of the road, or can they be within the road itself? Our plan had been to only have signals at intersections, but let me know if that is a bad assumption. If the signals are only encountered at the end of the road, then this would be better as an element of `Lane`, like `Lane.maneuvers`. In fact if there can be multiple signals for a given lane, e.g. if a left-turn arrow is represented separately from a green light, it would be good for each `Maneuver` to reference the signal which governs it.

Reply:
OpenDrive allows you to define signals anywhere in the map (i.e., inside intersections or not). Signals can also be defined in the middle of the road. There aren't restrictions in this sense. However, for the specific case of CARLA maps, traffic lights are always defined inside intersections.

As you say, right now, signals can only be retrieved from roads and intersections. Our plan is to have signals accessible from lanes and maneuvers also, but I think this would require some more work (e.g., check the lane validity of the signal and decide what to do with lane sections). Were were planning to add this functionality for a future PR.